new-backend/app/modules/langgraph_nodes/store_and_send.py

```
@@ -1,12 +1,26 @@
 
 def store_and_send(state):
     # to store data in vector db
     try:
-        save_to_vector_db({
-            **state
-        })
+        print(state)
+        try:
+            chunks = chunk_rag_data(state)
+        except KeyError as e:
+            raise Exception(f"Missing required data field for chunking: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)